hazelcast/src/test/java/com/hazelcast/scheduledexecutor/ScheduledExecutorServiceSlowTest.java

```diff
@@ -103,23 +103,26 @@ public void stats_longRunningTask_durable()
 
         String key = generateKeyOwnedBy(instances[1]);
 
-        ICountDownLatch latch = instances[0].getCountDownLatch("latch");
-        latch.trySetCount(6);
+        ICountDownLatch firstLatch = instances[0].getCountDownLatch("firstLatch");
+        firstLatch.trySetCount(2);
+
+        ICountDownLatch lastLatch = instances[0].getCountDownLatch("lastLatch");
+        lastLatch.trySetCount(6);
 
         IScheduledExecutorService executorService = getScheduledExecutor(instances, "s");
         IScheduledFuture future = executorService.scheduleOnKeyOwnerAtFixedRate(
-                new ICountdownLatchRunnableTask("latch"), key, 0, 10, SECONDS);
+                new ICountdownLatchRunnableTask("firstLatch", "lastLatch"), key, 0, 10, SECONDS);
 
-        Thread.sleep(12000);
+        firstLatch.await(12, SECONDS);
 
         instances[1].getLifecycleService().shutdown();
 
-        latch.await(70, SECONDS);
+        lastLatch.await(70, SECONDS);
         sleepSeconds(4); // Wait for run-cycle to finish before cancelling, in order for stats to get updated.
         future.cancel(false);
 
         ScheduledTaskStatistics stats = future.getStats();
-        assertEquals(6, stats.getTotalRuns());
+        assertEquals(6, stats.getTotalRuns(), 1);
```

Comment:
Thanks for the feedback @vbekiaris . I already cancel the task from the test, line 122. But with unfortunate timing and slow operations this is not reliable. Cancelling within the task is easier said than done, mainly because I don't have access on the future. Also gating the run with a branch wont work either because the stats get updated on a finally block on the outside `runnable` wrapper. I think without major changes, that's the best we can do, of course open to suggestions.